### PR TITLE
Library for baseline calculation

### PR DESCRIPTION
**Pull request by sarthak77**
Ref issue : https://github.com/hypertrace/hypertrace-alert-engine/issues/54

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -15,3 +15,4 @@ plugins {
 include(":gateway-service-api")
 include(":gateway-service-impl")
 include(":gateway-service")
+include("gateway-service-baseline-lib")
```
Comment (kotharironak) [inline settings.gradle.kts:18]:
```suggestion
include(":gateway-service-baseline-lib")
```

File: gateway-service-baseline-lib/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+dependencies {
+    api(project(":gateway-service-api"))
+
+    // Common utilities
+    implementation("org.apache.commons:commons-lang3:3.10")
+    implementation("org.apache.commons:commons-math:2.2")
+    implementation("com.google.protobuf:protobuf-java-util:3.17.3")
+
+    implementation("com.fasterxml.jackson.core:jackson-annotations:2.12.4")
```
Comment (rish691) [inline gateway-service-baseline-lib/build.gradle.kts:19]: is jackson needed?

File: gateway-service-baseline-lib/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    jacoco
+    id("org.hypertrace.jacoco-report-plugin")
+}
+
+tasks.test {
+    useJUnitPlatform()
+}
+
+dependencies {
+    api(project(":gateway-service-api"))
```
Comment (kotharironak) [inline gateway-service-baseline-lib/build.gradle.kts:12]: Cleanup un-used libraries.

File: gateway-service-baseline-lib/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+plugins {
+    `java-library`
+    jacoco
```
Comment (pavan-traceable) [inline gateway-service-baseline-lib/build.gradle.kts:3]: Which plugin publishes this jar ? Its needed right ?

Comment (rish691) [reply gateway-service-baseline-lib/build.gradle.kts:3]: yeah, @sarthak77 please add `id("org.hypertrace.publish-plugin")`

**Review by pavan-traceable — DISMISSED** [review]
Looks good to me.

File: gateway-service-baseline-lib/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    id("org.hypertrace.jacoco-report-plugin")
+    id("org.hypertrace.publish-plugin")
+}
+
+tasks.test {
+    useJUnitPlatform()
+}
+
+dependencies {
+    api(project(":gateway-service-api"))
```
Comment (kotharironak) [inline gateway-service-baseline-lib/build.gradle.kts:13]: It's a utility lib, so we have to remove the deps on this lib. For now, can you expose the method `private static Baseline getBaseline(double[] metricValueArray)` as public. As I guess, `hypetrace-alert-engine` doesn't understand gateway service `Value` proto.

Comment (kotharironak) [reply gateway-service-baseline-lib/build.gradle.kts:13]: I just saw that hypertrace-alert-engine is already using gateway-service-api.

File: gateway-service-baseline-lib/src/main/java/org/hypertrace/gateway/service/baseline/lib/BaselineCalculator.java [hunk 6]
```diff
@@ -18,7 +18,7 @@ public static Baseline getBaseline(List<Value> metricValues) {
     return getBaseline(values);
   }
 
-  private static Baseline getBaseline(double[] metricValueArray) {
+  public static Baseline getBaseline(double[] metricValueArray) {
```
Comment (kotharironak) [inline gateway-service-baseline-lib/src/main/java/org/hypertrace/gateway/service/baseline/lib/BaselineCalculator.java]: I just saw that hypertrace-alert-engine is already using gateway-service-api. In that case, we can keep it private.

**Comment by kotharironak** [comment]
As discussed here (https://github.com/hypertrace/hypertrace-alert-engine/issues/54#issuecomment-912651946), we are going ahead with the library fist, and once we add more functionality around the baseline, it can be wrapped via service if required.

**Review by kotharironak — APPROVED** [review]

**Review by pavan-traceable — APPROVED** [review]

**Review by rish691 — APPROVED** [review]

**Pull request merged**